modules/chapter1/pages/section1.adoc

```diff
@@ -18,27 +18,38 @@ Good machine learning scenarios often have the following common properties:
 
 Semi-Supervised Learning:: 
-Semi-Supervised learning is a machine learning algorithm that works between the supervised and unsupervised learning. So it uses both labelled and unlabelled data. It is particularly useful when obtaining labeled data is costly, time-consuming, or resource-intensive.
+Semi-Supervised learning is a machine learning algorithm that works between the supervised and unsupervised learning. It uses a small portion of  labelled and a large majority of unlabelled data. The labelled data is used to provide order to the learning problem while the unlabeled data is used to create a more generalized model by providing context. The upside is we are able to create more accurate and resilliant models by negating the caveats that exists for supervised and unsupervised learning.  It is particularly useful when obtaining labelled data is costly, time-consuming, or resource-intensive. One example of application are is Image and Speech analysis. 
 
 Reinforcement learning:: 
-It involves an agent learning to make decisions through trial and error interactions with an environment. The agent receives feedback in the form of rewards or penalties for its actions, guiding it towards achieving a goal by maximizing the rewards. E.g Autonomous driving, AlphaGo learning to play Go by playing against itself.
+In comparison to the previous groups of algorithms mentioned, reinforcement learning does not have the goal of grouping or classifying data. It tries to find the most optimal path. It involves an agent learning to make decisions through trial and error interactions with an environment. The agent receives feedback in the form of rewards or penalties for its actions, guiding it towards achieving a goal by maximizing the rewards. E.g Autonomous driving, AlphaGo learning to play Go by playing against itself. Examples of algorithms are Monte Carlo, Deep Q Networks(DQN), State-Action-Reward-State-Acction(SARSA) and others. 
 
+== Choosing your algorithm
+We have a plethora of mathematical algorithms that may, or may not, solve our problem and be applied on our data. But, how do you recognise the category of problem and the algorithm that can help you solve it?
+The clue lies with your outcome variable.
+ 
+In a classification problem, the outcome variable (the event you want to predict) is a category or class label. It can be binary, such as "Sick" or "Healthy", "Yes" or "No". They can be multi-class such as for image recognition when you want to identify more than 2 classes or labels. 
```

Comment:
Expanding the example a bit and adding a mention to multi-label problems.

```suggestion
In a classification problem, the outcome variable (the event you want to predict) is a category or class label. It can be binary, such as "Sick" or "Healthy", "Yes" or "No". The problem can be multi-class, such as for image recognition, when you want to identify more than two classes or labels, for example, dog breeds. Lastly, you can also have multi-label problems, where the model assigns more than one label to the same record. For example, a model might detect multiple labels in a single  music clip, e.g. rock, pop, 90s.
```